crates/miden-lib/asm/kernels/transaction/lib/input_note.masm

```diff
@@ -25,7 +25,7 @@ export.get_assets_info
     dup exec.memory::get_num_input_notes 
     # => [input_notes_num, note_index, note_index]
     
-    lt assert.err=ERR_NOTE_INDEX_OUT_OF_BOUNDS
+    u32lt assert.err=ERR_INPUT_NOTE_INDEX_OUT_OF_BOUNDS
```

Comment:
To use `u32lt` we need to be sure that both values on the stack are `u32` values. I don't think we know this about `note_index`, and so, I would write this as:
```
u32assert2.err=ERR_INPUT_NOTE_INDEX_OUT_OF_BOUNDS
u32lt assert.err=ERR_INPUT_NOTE_INDEX_OUT_OF_BOUNDS
```